### PR TITLE
Streamline upgrade docs

### PR DESCRIPTION
**Pull request by joestringer**
Reviewers: review commit-by-commit, the commit logs have more detail.

This PR attempts to simplify and streamline the upgrade process by eliminating paths and old documentation. It also consolidates some pieces of information to help developers avoid forgetting to update it after a minor release version bump. Finally, the version-specific notes are updated.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10505)

### CODE REVIEW

**Comment by joestringer** [comment]
test-docs-please

File: Documentation/install/upgrade.rst [hunk 1]
```diff
@@ -279,6 +241,16 @@ latest ``1.1.y`` release before subsequently upgrading to ``1.2.z``.
...
 +-----------------------+-----------------------+-----------------------+-------------------------+---------------------------+
+| ``1.6.x``             | ``1.6.6``             | Not required          | Minimal to None         | Clients must reconnect[1] |
++-----------------------+-----------------------+-----------------------+-------------------------+---------------------------+
+| ``1.6.x``             | ``1.6.7``             | Required              | Minimal to None         | Clients must reconnect[1] |
++-----------------------+-----------------------+-----------------------+-------------------------+---------------------------+
+| ``1.6.x``             | ``1.7.y``             | Required              | Minimal to None         | Clients must reconnect[1] |
++-----------------------+-----------------------+-----------------------+-------------------------+---------------------------+
+| ``1.7.0``             | ``1.7.1``             | Required              | Minimal to None         | Clients must reconnect[1] |
++-----------------------+-----------------------+-----------------------+-------------------------+---------------------------+
```
Comment (aanm) [inline Documentation/install/upgrade.rst]: Maybe we should switch the column name "``DaemonSet`` upgrade" to something else just to give the impression that the k8s descriptor needs to be re-generated.

**Review by errordeveloper — COMMENTED** [review]
Nice! Just out of curiosity, how long ago did the DNS poller get deprecated?

**Review by errordeveloper — APPROVED** [review]

**Comment by joestringer** [comment]
@errordeveloper that's a good question. #8604 used to plan deprecation for v1.6 and removal in v1.7, though I didn't yet dig into whether we actually declared the DNS poller dead; so I've reworked that issue description to plan deprecation in v1.8 and drop the code in v1.9.

That said we've been encouraging users to switch from the poller mode to the proxy mode since v1.5 or so I believe. I'm not aware of any users enabling the mode.

**Comment by errordeveloper** [comment]
@joestringer thanks, sounds like a good call to soft-deprecate it by removing the docs!

**Review by pchaigno — APPROVED** [review]

**Comment by joestringer** [comment]
test-docs-please

**Pull request merged**